drivers/shared/validators/validators.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package validators
```

Comment:
I feel like pulling this stuff into a new shared package was probably the biggest "design" choice I made in this. There didn't feel like another obvious place to put shared driver validation code, but perhaps I missed something. (also, not the point of this package, but it could probably be more broadly useful)

Reply:
I think in most cases any validation is driver specific. Since this is shared between exec and raw_exec, seems like a reasonably choice to put it here.